server/cmd/mmctl/commands/webhook_test.go

```diff
@@ -92,26 +140,57 @@ func (s *MmctlUnitTestSuite) TestListWebhookCmd() {
 
 		s.client.
 			EXPECT().
-			GetIncomingWebhooksForTeam(context.TODO(), teamID, 0, 100000000, "").
-			Return([]*model.IncomingWebhook{&mockIncomingWebhook}, &model.Response{}, nil).
+			GetIncomingWebhooksForTeam(context.TODO(), teamID, 0, 200, "").
+			Return(mockIncomingWebhooksPage1, &model.Response{}, nil).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetIncomingWebhooksForTeam(context.TODO(), teamID, 1, 200, "").
+			Return(mockIncomingWebhooksPage2, &model.Response{}, nil).
 			Times(1)
 
 		s.client.
 			EXPECT().
-			GetOutgoingWebhooksForTeam(context.TODO(), teamID, 0, 100000000, "").
-			Return([]*model.OutgoingWebhook{&mockOutgoingWebhook}, &model.Response{}, nil).
+			GetIncomingWebhooksForTeam(context.TODO(), teamID, 2, 200, "").
+			Return(nil, &model.Response{}, nil).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetOutgoingWebhooksForTeam(context.TODO(), teamID, 0, 200, "").
+			Return(mockOutgoingWebhooksPage1, &model.Response{}, nil).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetOutgoingWebhooksForTeam(context.TODO(), teamID, 1, 200, "").
+			Return(mockOutgoingWebhooksPage2, &model.Response{}, nil).
+			Times(1)
+
+		s.client.
+			EXPECT().
+			GetOutgoingWebhooksForTeam(context.TODO(), teamID, 2, 200, "").
+			Return(nil, &model.Response{}, nil).
 			Times(1)
 
 		err := listWebhookCmdF(s.client, &cobra.Command{}, []string{teamID})
 		s.Require().Nil(err)
-		s.Len(printer.GetLines(), 2)
 		s.Len(printer.GetErrorLines(), 0)
-		s.Require().Equal(&mockIncomingWebhook, printer.GetLines()[0])
-		s.Require().Equal(&mockOutgoingWebhook, printer.GetLines()[1])
+		if s.Len(printer.GetLines(), 500) {
+			for i := 0; i < 250; i++ {
+				s.Require().Equal(&mockIncomingWebhook, printer.GetLines()[i])
+			}
+			for i := 250; i < 500; i++ {
+				s.Require().Equal(&mockOutgoingWebhook, printer.GetLines()[i])
+			}
+		}
 	})
 
 	s.Run("Unable to list webhooks", func() {
 		printer.Clean()
+		s.mockCtrl = gomock.NewController(s.T())
+		s.client = mocks.NewMockClient(s.mockCtrl)
```

Comment:
Why do we need to create a new controller&client here?

Reply:
I've tried to run the test without the reset and they're working fine. I think this only happens when an assertion fails, which is definitely bad (errors should not bleed into other tests) but we don't need to reset before the test if no errors are triggered

Reply:
You're right, @mgdelacroix, the need to reset the test is only made apparent when an error occurs: would you agree that means we should reset all the time to keep the tests isolated? (One could also imagine false positives from overlapping expectations between tests.)

Maybe there's somewhere we can do this more holistically for all tests?

Reply:
Sorry I missed this. I agree that we should do everything we can to isolate all tests; checking the code, it seems that we're already resetting the controller and the mocked client on every test, but it doesn't take into account the subtests. It might be just a matter of implementing a `SetupSubTest` method that performs that same reset as part of each `s.Run`